src/main/java/Ui.java

```diff
@@ -0,0 +1,183 @@
+import java.util.Scanner;
+
+/**
+ * Ui class is the main processor for users inputs,
+ * Ui takes in user inputs, categorises it and stores it.
+ */
+public class Ui {
+
+    private final Scanner scanner;
+    private final Parser parser;
+
+    public Ui(){
+        scanner = new Scanner(System.in);
+        parser = new Parser();
+    }
+
+    public void deal(){
+        for (int i = 0; i < 100000; i++) {
+
+            String input = this.scanner.nextLine();
+
+            if (!parser.isBye(input)) {
+
+                if (!parser.isList(input) && !parser.isFind(input)) {
+
+                    if (!parser.isDone(input) && !parser.isDelete(input)) {
+
+                        if (parser.isTodo(input) || parser.isEvent(input)
+                                || parser.isDeadline(input)) {
```

Comment:
Do you think you can abstract this expression? This is rather complicated to look at if the person is looking at the code for the first time. You might consider:

boolean isInputTodo = parser.isTodo(input);
boolean isInputEvent = parser.isEvent(input);
boolean isInputDeadline = parser.isDeadline(input);

This is to further enhance the readability of the code, but this is just a small comment.